deploy/scripts/src/mobile/utils/test-client.ts

```diff
@@ -1,23 +1,54 @@
 import http, { type Response } from 'k6/http'
-import { buildUrl } from './url'
+import { buildBackendUrl, buildUrl } from './url'
 import { config } from './config'
 
 export function postTestClientStart (): Response {
   return http.post(
     buildUrl('start', config.testClientExecuteUrl),
-    JSON.stringify({ frontendUri: config.frontendUrl }),
+    JSON.stringify({
+      target: config.backendUrl,
+      frontendUri: config.frontendUrl
+    }),
     {
       tags: { name: 'Post request to authorize URL' },
       headers: { 'Content-Type': 'application/json' }
     }
   )
 }
 
-export function parseTestClientResponse (response: Response, location: 'WebLocation' | 'ApiLocation'): string {
-  const url = response.json(location)
+export function parseTestClientResponse (
+  response: Response,
+  location: 'WebLocation' | 'ApiLocation'
+): string {
+  let url
+  if (location === 'ApiLocation') {
+    const apiLocation = response.json(location) as string
+    url = parseApiLocation(apiLocation)
```

Comment:
By putting `parseApiLocation` before the `if (typeof url !== 'string') {` check, we're changing the behaviour of the error handling of this function. I think the `if (location === 'ApiLocation')` check should come after the `if (typeof url !== 'string')`, i.e. something like this:
```
const url = response.json(location)
if (typeof url !== 'string') {
    throw new Error('Failed to parse URL from response')
}
if (location === 'ApiLocation') {
  return parseApiLocation(url)
} else {
  return url
}
```